src/modules/import/domain/import-workflow.ts

```diff
@@ -255,96 +247,90 @@ export class ImportWorkflow {
-            // We should not use the CreationService here
-            const savedPersonWithPersonenkontext: DomainError | PersonPersonenkontext =
-                await this.personenkontextCreationService.createPersonWithPersonenkontexte(
-                    permissions,
-                    importDataItem.vorname,
-                    importDataItem.nachname,
-                    createPersonenkontexte,
-                );
-            if (!(savedPersonWithPersonenkontext instanceof DomainError)) {
-                this.logger.info(
-                    `System hat einen neuen Benutzer ${savedPersonWithPersonenkontext.person.referrer} (${savedPersonWithPersonenkontext.person.id}) angelegt.`,
-                );
-            } else {
-                this.logger.info(
-                    `System hat versucht einen neuen Benutzer für ${importDataItem.vorname} ${importDataItem.nachname} anzulegen. Fehler: ${savedPersonWithPersonenkontext.message}`,
-                );
-            }
-            savedPersonenWithPersonenkontext.push(savedPersonWithPersonenkontext);
+    public async downloadFile(importvorgangId: string, permissions: PersonPermissions): Promise<Result<Buffer>> {
+        const permissionCheckError: Option<DomainError> = await this.checkPermissions(permissions);
+        if (permissionCheckError) {
+            return {
+                ok: false,
+                error: permissionCheckError,
+            };
         }
-        /* eslint-disable no-await-in-loop */
 
-        //Save Benutzer + Passwort in the Liste
-        savedPersonenWithPersonenkontext.forEach((personPersonenkontext: DomainError | PersonPersonenkontext) => {
-            if (!(personPersonenkontext instanceof DomainError)) {
-                const klasse: OrganisationByIdAndName | undefined = klassenByIDandName.find(
-                    (klasseByIDandName: OrganisationByIdAndName) =>
-                        personPersonenkontext.personenkontexte.some(
-                            (pk: Personenkontext<true>) => pk.organisationId === klasseByIDandName.id,
-                        ),
-                );
+        const importVorgang: Option<ImportVorgang<true>> = await this.importVorgangRepository.findById(importvorgangId);
+        if (!importVorgang) {
+            return {
+                ok: false,
+                error: new EntityNotFoundError('ImportVorgang', importvorgangId),
+            };
+        }
+        //Will never happen
+        if (!importVorgang.organisationId) {
+            return {
+                ok: false,
+                error: new ImportDomainError('ImportVorgang is missing an organisazion id', importvorgangId),
+            };
+        }
+        if (!importVorgang.rolleId) {
+            return {
+                ok: false,
+                error: new ImportDomainError('ImportVorgang is missing a rolle id', importvorgangId),
+            };
+        }
 
```

Comment:
Bitte auch als warning loggen. Wenn wir das dann in Prod sehen wissen wir das irgendwas deutlich falsch gelaufen ist wenn ein Zustand der logisch nicht erreichbar sein sollte erreicht wurde.